File: ProjetoSpaceNowAngular/src/app/Models/andar.model.ts
export class PredioModel {
  predioId: number = 0;
  nome: string = '';
  apelido?: string;
  endereco?: string;
  ids: string[] = [];
}

export class AndarModel {
  andarId: number = 0;
  apelido: string = '';
  predio: PredioModel = new PredioModel();
  ids: string[] = [];
}

File: ProjetoSpaceNow/ProjetoSpaceNow.Api/Models/AndarModel.cs
using Supabase.Postgrest.Attributes;

namespace ProjetoSpaceNow.Api.Models
{
    [Table("andar")]
    public class AndarModel : BaseEntity
    {

        [Column("apelido")]
        public string Apelido { get; set; }

        [Column("predio_id")]
        public int PredioId { get; set; }

        [Reference(typeof(PredioModel))]
        public PredioModel Predio { get; set; }

        [Reference(typeof(SalaModel))]
        public ICollection<SalaModel> Salas { get; set; }
    }
}
